core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -2164,7 +2164,10 @@ private[spark] class BlockManager(
     diskBlockManager.stop()
     rpcEnv.stop(storageEndpoint)
     blockInfoManager.clear()
-    memoryStore.clear()
+    // The memoryManager may be null if the driver plugin fails to initialize
```

Comment:
Should we fix it the other way around? Do not break Spark initialization if the plugin fails to initialize.

Reply:
I'm not sure if we should keep starting spark context if driver plguin fail to initialize. Before we have fixed an issue for executor plugin [SPARK-40320](https://issues.apache.org/jira/browse/SPARK-40320) , by making exexutor fast fail when it fails to initialize. So this pr leave driver fast fail and just drops the noisy error message.

Reply:
I see

Reply:
Agree, if driver plugin fails to initialize, we should fail fast - the functionality could (and tends to be) critical for the spark application: for example, remote shuffle service initialization.